src/main/java/model/structures/GraphImpl.java

```diff
@@ -0,0 +1,144 @@
+package model.structures;
+
+import model.generator.ModelGenerator;
+
+import java.util.*;
+
+public class GraphImpl implements Graph {
+
+    private HashMap<Integer, Node> nodes;
+    private int[][] adjacencyMatrix;
+    private Random randomForAdjacencyMatrix;
+    private int edgeWageSum;
+    private int graphSize;
+    private int iterationStep;
+    private final long defaultSeedForWages = 8064995938733697569L;
+
+    public GraphImpl(long seedForWages, long seedForHabitats, int edgeWageSum, int minimalNumberOfHabitats, int maximumNumberOfHabitats, int graphSize, int infectingParameter)
```

Comment:
Konstruktor nie powinien mieć tylku parametrów. Lepiej użyć Buildera

Reply:
I tak te wszystkie parametry będzie trzeba przekazać do tej klasy. Moim zdaniem sprowadzi się to do dopisania pakietu dodatkowych setterów, które są zbędne, a i tak będziesz musiała to do buildera jakoś przekazać